ipet/parsing/Solver.py

```diff
@@ -565,6 +567,26 @@ def extractOptionalInformation(self, line : str):
         """
         self.extractPath(line)
         self.extractMoreData(line)
+        self.extractPrimalHeuristics(line)
+
+    def extractPrimalHeuristics(self, line: str):
+        if line.startswith('Primal Heuristics'):
+            self.primalHeuristicsStatsArePrinted = True
+        elif not line.startswith(' '):
+            self.primalHeuristicsStatsArePrinted = False
+        elif self.primalHeuristicsStatsArePrinted:
+            name = re.compile("(\s+)(\w+\s?\w+)(\s*)").match(line).groups()[1]
+            execTimeExpr = re.compile("\s+\w+\s?\w+\s*:\s+(\S+)")
+            setupTimeExpr = re.compile("\s+\w+\s?\w+\s*:\s+\d+.\d+\s+(\S+)")
+            callsExpr = re.compile("\s+\w+\s?\w+\s*:\s+\d+.\d+\s+\d+.\d+\s+(\S+)")
+            foundExpr = re.compile("\s+\w+\s?\w+\s*:\s+\d+.\d+\s+\d+.\d+\s+\d+\s+(\S+)")
+            bestExpr = re.compile("\s+\w+\s?\w+\s*:\s+\d+.\d+\s+\d+.\d+\s+\d+\s+\d+\s+(\S+)")
```

Comment:
Maybe define these expressions outside, so that you don't have to compile them in every line you parse. 
But i'm unsure if that is relevant, since you probably don't parse primalheuristicstats in very few lines.

Reply:
yeah I think it improves the readability if for the not major datakeys the expressions are not listed above- to have some kind of distinction in the code.
Bout as you said it comes at the cost of performance (event though it should be notable).
But I don't have a strong opinion on that.